src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/RepositoryCloneManager.cs

```diff
@@ -95,6 +111,60 @@ public class RepositoryCloneManager : IRepositoryCloneManager
         return path;
     }
 
+    public async Task<NativePath> PrepareClone(
+        SourceMapping mapping,
+        IReadOnlyCollection<string> remoteUris,
+        IReadOnlyCollection<string> requestedRefs,
+        string checkoutRef,
+        CancellationToken cancellationToken)
+    {
+        if (remoteUris.Count == 0)
+        {
+            throw new ArgumentException("No remote URIs provided to clone");
+        }
+
+        var refsToVerify = new HashSet<string>(requestedRefs);
+
+        NativePath path = null!;
+        foreach (string remoteUri in remoteUris)
+        {
+            // Path should be returned the same for all invocations
+            // We checkout a default ref
+            path = await PrepareCloneInternal(remoteUri, mapping.Name, cancellationToken);
+
+            // Verify that all requested commits are available
+            foreach (var commit in refsToVerify.ToArray())
+            {
+                try
+                {
+                    await _localGitRepo.GetShaForRefAsync(path, commit);
+                    refsToVerify.Remove(commit);
+                }
+                catch
+                {
+                    // Ref not found yet, let's try another remote
+                    continue;
+                }
+
+                if (refsToVerify.Count == 0)
+                {
+                    _logger.LogDebug("All requested refs ({refs}) found in {repo}", string.Join(", ", requestedRefs), path);
```

Comment:
Technically, this may not be true. We may find some refs in one remote, and other refs in another.

Reply:
This logs the path to the clone, not the remote